packages/grid/src/vaadin-grid.js

```diff
@@ -653,13 +653,28 @@ class Grid extends ElementMixin(
 
     const cell = document.createElement(tagName);
     cell.id = slotName.replace('-content-', '-');
-    cell.setAttribute('tabindex', '-1');
     cell.setAttribute('role', tagName === 'td' ? 'gridcell' : 'columnheader');
 
     const slot = document.createElement('slot');
     slot.setAttribute('name', slotName);
 
```

Comment:
This PR seems to do the job even without the additional wrappers if you apply the `role` and `tabindex` to the `<slot>` element directly. If that's an option, it would be a less dramatic change overall.

Tested by adding the following here:
```js
slot.setAttribute('role', 'button');
slot.setAttribute('tabindex', '-1');
```

https://user-images.githubusercontent.com/1222264/183872638-b2d0a8c4-dc38-4c48-a9d0-acdffc75af3a.mp4

Reply:
One issue I see with this is that `role` is not a valid attribute on `slot` elements, so it'll be flagged by a11y checkers (and possibly general html checkers too). I could imagine `tabindex` getting flagged too, but setting `tabindex=-1` on an element that, by definition, cannot be focused, is hardly a big deal.

Even so, I would consider these as acceptable compromises, if it generally improves a11y in practice, however. So I'd say let's test that solution with NVDA and JAWS (on Chrome and FF), as well as iOS VoiceOver, and if it's better or at least no worse than before

Reply:
Discussed this once again and decided to not use `role` on the `<slot>` elements.